src/rgw/rgw_op.cc

```diff
@@ -5202,9 +5202,13 @@ void RGWListBucketMultiparts::execute()
 
 void RGWGetHealthCheck::execute()
 {
-  if (! g_conf->rgw_healthcheck_disabling_path.empty() &&
-      ::access(g_conf->rgw_healthcheck_disabling_path.c_str(), F_OK )) {
-    op_ret = -ERR_SERVICE_UNAVAILABLE;
+  if (!g_conf->rgw_healthcheck_disabling_path.empty()) {
+    if (::access(g_conf->rgw_healthcheck_disabling_path.c_str(), F_OK) == 0) {
+      /* path exists */
+      op_ret = -ERR_SERVICE_UNAVAILABLE;
+    } else {
```

Comment:
No reason but for to be more explicit on the path existence and access failures. Corrected it now.